pennylane/interfaces/batch/__init__.py

```diff
@@ -16,14 +16,59 @@
 capabilities with different machine learning libraries.
 """
 # pylint: disable=import-outside-toplevel,too-many-arguments,too-many-branches
+import contextlib
 from functools import wraps
+import itertools
 
 from cachetools import LRUCache
 import numpy as np
 
 import pennylane as qml
 
 
+INTERFACE_NAMES = {
+    "NumPy": (None,),
+    "Autograd": ("autograd", "numpy"),  # for backwards compatibility
+    "PyTorch": ("torch", "pytorch"),
+    "TensorFlow": ("tf", "tensorflow"),
+}
+"""dict[str, str]: maps allowed interface strings to the name of the interface"""
+
+SUPPORTED_INTERFACES = list(itertools.chain(*INTERFACE_NAMES.values()))
+
+
+@contextlib.contextmanager
+def set_shots(device, shots):

```

Comment:
does this allow the number of shots to be changed for each execution in a batch or does it change all executions to a single new number of shots?

Reply:
The latter 🙂 

Now that you mention it though, I think there is a very good use-case for the _former_ as well 🤔